### ISSUE TITLE
Documentation request: pytest.mark.parametrize + brackets in dependency name

### ISSUE THREAD

**Issue by jamesbraza** [RKrahl/pytest-dependency#49]
I am making this issue because I believe the use of [`pytest.mark.parametrize`](https://docs.pytest.org/en/stable/parametrize.html) requires knowledge not currently in `pytest-dependency`'s docs: bracket syntax used in the name of parametrized tests.

I was trying to use `pytest-dependency` and `pytest.mark.parametrize` together, and couldn't figure it out.
- Searching in the `pytest-dependency` docs for "parametrize" --> no results
- Searching through the issues --> another user had the same problem last summer here https://github.com/RKrahl/pytest-dependency/issues/45
    - I was able to follow that issue to realize I needed to use brackets in the dependency's name.

Here's some sample code:

```python
import pytest

class TestOneTwo:
    """Bunch of tests grouped in a class."""

    PARAM_ONE = 1001
    PARAM_TWO = 1002

    @pytest.mark.parametrize("foo", [PARAM_ONE, PARAM_TWO])
    @pytest.mark.dependency()
    def test_one(self, foo: int) -> None:
        assert isinstance(foo, int)

    @pytest.mark.dependency(depends=[f"TestOneTwo::test_one[{PARAM_ONE}]"])
    def test_two_option_1(self) -> None:
        pass

    @pytest.mark.dependency(depends=[f"test_one[{PARAM_ONE}]"], scope="class")
    def test_two_option_2(self) -> None:
        pass
```

I think the docs should address that use of `pytest.mark.parametrize` requires brackets `[x]` in the `depends` field to address parametrized tests.

**Comment by RKrahl** [comment]
Actually, it is mentioned in the documentation. There is a dedicated [section on the names of tests](https://pytest-dependency.readthedocs.io/en/stable/names.html) and how they need to be referenced in the `dependency` marker. It states that the default is the pytest node id and that: “If the function or method is parameterized, the parameter values, separated by minus ‘-’, in square brackets ‘[]’ are appended to the node id.”

Note that the way pytest creates node ids for parametrized tests may become complicated, in particular if the parameter values do not have obvious string representations or if one set of parameter values appears more then ones. Documenting all the intricate details is probably out of scope of the pytest-dependency documentation. I rather tend to resort recommending trying it out in the given case by calling pytest with the `–verbose` command line option. This is also already mentioned in the `Names` section but might get elaborated in the context of #41.

**Comment by jamesbraza** [comment]
Just read through the Names page, and I see that it has lots of information on `pytest.mark.parametrize`.  I realized that when searching:
- "parametrize": Names page doesn't come up
- "parametrized" or "pytest.mark.parametrize": multiple results mentioning parametrization comes up

I didn't know Read the Docs had such an exact search... otherwise I wouldn't have made this issue.

I also didn't know one can actually use `pytest-dependency` inside of `pytest` parametrizations, or even account for dependency on all instances of a parametrized test.  Turns out usage with `pytest.mark.parametrize` is actually well documented, and flexible!

I think a debugging page would be nice for sure.  You're doing great work here @RKrahl !
